source/connection.cc

```diff
@@ -209,6 +211,21 @@ RestClient::Connection::SetKeyPath(const std::string& keyPath) {
   this->keyPath = keyPath;
 }
 
+void
+RestClient::Connection::SetProxy(const std::string& uriProxy) {
+  if (uriProxy.empty())
+    return;
+
+  std::string uriProxyUpper = uriProxy;
+  std::transform(uriProxyUpper.begin(), uriProxyUpper.end(),
+    uriProxyUpper.begin(), ::toupper);
+
+  if (uriProxyUpper.compare(0, 4, "HTTP") != 0)
+    this->uriProxy = "http://" + uriProxy;
+  else
+    this->uriProxy = uriProxy;
```

Comment:
I can add a test similar to "TestProxy" and provide a proxy address that begins with "http" or "https". I think that the proxy I already used in that test is a one using SSL but I might be wrong.